Makefile

```diff
@@ -1,41 +1,56 @@
-install-root-dependencies:
+help:                                                                           ## shows this help
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_\-\.]+:.*?## / {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
+
+install-root-dependencies:                                                      ## install dependencies for the library itself
 	composer update
 
-install-coding-standard-dependencies:
+install-coding-standard-dependencies:                                           ## install dependencies for coding-standard checks tooling
 	cd tools/php-cs-fixer && composer update --ignore-platform-req php
 	cd tools/php-codesniffer && composer update
 
-install-static-analysis-dependencies:
+install-benchmark-dependencies:                                                 ## install dependencies for benchmark tooling
+	cd tools/phpbench && composer install
+
+install-static-analysis-dependencies:                                           ## install dependencies for static analysis tooling
 	cd tools/psalm && composer update
 
-install-unit-tests-dependencies:
+install-unit-tests-dependencies:                                                ## install dependencies for the test suite
 	cd tools/phpunit && composer update
 
-install: install-root-dependencies install-coding-standard-dependencies install-static-analysis-dependencies install-unit-tests-dependencies
+install: install-root-dependencies install-coding-standard-dependencies install-benchmark-dependencies install-static-analysis-dependencies install-unit-tests-dependencies ## install all dependencies for a development environment
 
-coding-standard-fix:
+coding-standard-fix:                                                            ## apply automated coding standard fixes
 	./tools/php-cs-fixer/vendor/bin/php-cs-fixer fix --config=tools/php-cs-fixer/.php_cs.dist.php
 	./tools/php-codesniffer/vendor/bin/phpcbf --basepath=. --standard=tools/php-codesniffer/.phpcs.xml
 
-coding-standard-check:
+coding-standard-check:                                                          ## check coding-standard compliance
 	./tools/php-cs-fixer/vendor/bin/php-cs-fixer fix --config=tools/php-cs-fixer/.php_cs.dist.php --dry-run
 	./tools/php-codesniffer/vendor/bin/phpcs --basepath=. --standard=tools/php-codesniffer/.phpcs.xml
 
-static-analysis:
+benchmarks:                                                                     ## run benchmarks
+	./tools/phpbench/vendor/bin/phpbench run --config tools/phpbench/phpbench.json
+
+create-benchmark-reference:                                                     ## run benchmarks, mark current run as "reference"
+	./tools/phpbench/vendor/bin/phpbench run --config tools/phpbench/phpbench.json --tag=benchmark_reference
+
+compare-benchmark-to-reference:                                                 ## run benchmarks, compare result to the "reference" run
```

Comment:
Perhaps better docs needed, but the idea is that you run `create-benchmark-reference` first :)